nixos/modules/services/misc/gitlab.nix

```diff
@@ -598,21 +573,38 @@ in {
 
         # JSON is a subset of YAML
         ln -fs ${pkgs.writeText "gitlab.yml" (builtins.toJSON gitlabConfig)} ${cfg.statePath}/config/gitlab.yml
-        ln -fs ${pkgs.writeText "database.yml" databaseYml} ${cfg.statePath}/config/database.yml
-        ln -fs ${pkgs.writeText "secrets.yml" secretsYml} ${cfg.statePath}/config/secrets.yml
         ln -fs ${pkgs.writeText "unicorn.rb" unicornConfig} ${cfg.statePath}/config/unicorn.rb
 
         chown -R ${cfg.user}:${cfg.group} ${cfg.statePath}/
         chmod -R ug+rwX,o-rwx+X ${cfg.statePath}/
 
+        cat > ${cfg.statePath}/config/secrets.yml << EOF
+        production:
+          secret_key_base: $(cat "${cfg.secrets.secret}")
```

Comment:
Does this really work? I think you should do:

```
nix-repl> "${toString ./timeout.txt}"
"/home/ielectric/timeout.txt"

nix-repl> "${./timeout.txt}"
"/nix/store/qvggx3asvcz26r9wpclpy67f0k5zryg6-timeout.txt"
```

Reply:
See https://stackoverflow.com/a/43850372/133235 as a reference

Reply:
I'm sorry, I don't understand exactly where you think it's going wrong. Can you point me at the exact spot? Here's an example of the rendered preStart-script:

    cat > /var/gitlab/state/config/secrets.yml << EOF
    production:
      secret_key_base: $(cat "/run/keys/secret")
      otp_key_base: $(cat "/run/keys/otp")
      db_key_base: $(cat "/run/keys/db")
      jws_private_key: $(/nix/store/dc4f9mfgbwb0wygwk7qw5b0s0mv7kcml-jq-1.5/bin/jq -Rs . "/run/keys/jws")
    EOF

Reply:
This is with services.gitlab.secrets.db set to "/run/keys/db" and so on.

Reply:
The problem is, someone can specify it as `/run/keys/db` instead of `"/run/keys/db"` and it would get copied to the store if you don't call `toString` first.

Reply:
Good point! That's actually a fair argument against using path at all...

Reply:
0ba5f49 incorporates your suggestion, thank you. :)

Reply:
We could come up with a type like `nonStorePath` meaning it would validate the value is not in store but is a path.

Reply:
I think that would actually be pretty handy!

Reply:
@domenkozar Do you still want `nonStorePath` for this PR?